src/darwin/Framework/CHIPTests/MTRDeviceTests.m

```diff
@@ -1402,6 +1402,45 @@ - (void)test017_TestMTRDeviceBasics
         [subscriptionDroppedExpectation fulfill];
     };
 
+    // Before resubscribe, first test write failure and expected value effects
+    NSNumber * testEndpointID = @(1);
+    NSNumber * testClusterID = @(8);
+    NSNumber * testAttributeID = @(10000);
+    XCTestExpectation * expectedValueReportedExpectation = [self expectationWithDescription:@"Expected value reported"];
+    XCTestExpectation * expectedValueRemovedExpectation = [self expectationWithDescription:@"Expected value removed"];
+    delegate.onAttributeDataReceived = ^(NSArray<NSDictionary<NSString *, id> *> * attributeReport) {
+        for (NSDictionary<NSString *, id> * attributeDict in attributeReport) {
+            MTRAttributePath * attributePath = attributeDict[MTRAttributePathKey];
+            XCTAssertNotNil(attributePath);
+            if ([attributePath.endpoint isEqualToNumber:testEndpointID] && [attributePath.cluster isEqualToNumber:testClusterID] &&
+                [attributePath.attribute isEqualToNumber:testAttributeID]) {
+                id data = attributeDict[MTRDataKey];
+                if (data) {
+                    [expectedValueReportedExpectation fulfill];
+                } else {
+                    [expectedValueRemovedExpectation fulfill];
+                }
+            }
+        }
+    };
+
+    NSDictionary * writeValue = [NSDictionary
+        dictionaryWithObjectsAndKeys:@"UnsignedInteger", @"type", [NSNumber numberWithUnsignedInteger:200], @"value", nil];
+    [device writeAttributeWithEndpointID:testEndpointID
+                               clusterID:testClusterID
+                             attributeID:testAttributeID
+                                   value:writeValue
+                   expectedValueInterval:@(20)
+                       timedWriteTimeout:nil];
+
+    // expected value interval is 20 but expect it get reverted immediately as the write fails
+    [self waitForExpectations:@[ expectedValueReportedExpectation, expectedValueRemovedExpectation ] timeout:5 enforceOrder:YES];
```

Comment:
This timeout is 5s.

Should we perhaps be using an expected value interval longer than 5s to really test this?

Reply:
I've changed the expectedValueInterval to 20000 / 20 seconds.